File: app/src/main/java/com/yatatsu/passiveviewsample/ui/user/UserController.java
package com.yatatsu.passiveviewsample.ui.user;

import com.yatatsu.passiveviewsample.data.api.GitHubService;
import com.yatatsu.passiveviewsample.data.model.User;
import com.yatatsu.passiveviewsample.ui.base.Controller;

import rx.functions.Action1;


public class UserController extends Controller<UserScreen> {

    private final GitHubService gitHubService;
    private String username;

    public UserController(GitHubService gitHubService) {
        this.gitHubService = gitHubService;
    }

    void onCreateScreen(String username) {
        this.username = username;
        register(gitHubService.getUser(username)
                .subscribe(
                        new Action1<User>() {
                            @Override
                            public void call(User user) {
                                getScreen().showUserImage(user.avatarUrl);
                                getScreen().showUserName(user.name);
                            }
                        }, new Action1<Throwable>() {
                            @Override
                            public void call(Throwable throwable) {
                                throwable.printStackTrace();
                                getScreen().showErrorDialog(throwable.getMessage());
                            }
                        }));
    }

    void onClickReposButton() {
        getScreen().navigateToReposScreen(username);
    }

    void onDestroyScreen() {
        unSubscribe();
    }
}

File: app/src/main/java/com/yatatsu/passiveviewsample/ui/user/UserController.kt
package com.yatatsu.passiveviewsample.ui.user

import com.yatatsu.passiveviewsample.data.api.GitHubService
import com.yatatsu.passiveviewsample.ui.ApiErrorPresenter
import com.yatatsu.passiveviewsample.ui.base.Controller


class UserController(private val gitHubService: GitHubService) : Controller<UserScreen>() {
  private var username: String? = null

  fun onCreateScreen(username: String?) {
    this.username = username
    register(gitHubService.getUser(username!!).subscribe(
        { user ->
          screen?.showUserImage(user.avatarUrl!!)
          screen?.showUserName(user.name!!)
        }) { throwable ->
      throwable.printStackTrace()
      screen?.showErrorDialog(
          ApiErrorPresenter.getErrorMessage(throwable))
    })
  }

  fun onClickReposButton() {
    screen?.navigateToReposScreen(username!!)
  }

  fun onDestroyScreen() {
    unSubscribe()
  }
}
